synapse/storage/databases/main/events.py

```diff
@@ -1954,19 +1954,19 @@ def _handle_redaction(self, txn, redacted_event_id):
         )
 
     def _store_room_topic_txn(self, txn, event):
```

Comment:
Please add some type hints while we're here!

I'm assuming an `event` is `EventBase`? Doesn't that always have a `content` attribute?

Reply:
Good idea!

`content` is defined as `DictProperty("content")` so I'm not sure. It depends on whether we can get here with an event that's missing `content`?

```py
>>> event_from_pdu_json({"type": "m.room.message", "depth": 1, "content": {"body": "hi"}}, RoomVersions.V7).content
{'body': 'hi'}
>>> event_from_pdu_json({"type": "m.room.message", "depth": 1}, RoomVersions.V7).content
[...]
AttributeError: '<class 'synapse.events.FrozenEventV3'>' has no 'content' property
```

Reply:
Huh fair, I suspect that isn't really possible, but I'm not sure, might be worth asking others? I believe in other places where we grab from content we just assume it exists.

Reply:
The spec says that `content` is a required field for every room version, so we shouldn't get here with events that are missing `content`s. Let's remove the `hasattr` check.